### DISCUSSION TITLE
Unable to parse text properly

### DISCUSSION THREAD

**Discussion by SridharVoleti** [jsvine/pdfplumber#744] [Ask for help with specific PDFs]
I am trying to extract text from a PDF file and I get an output of the entire text without a space in it. How do i solve this issue? Below is a snapshot of page 22 of the [book](https://drive.google.com/file/d/1jDLMYiEvY0z_YsxOrTBjJdJz0fFZtk4V/view?usp=sharing)
![image](https://user-images.githubusercontent.com/58073176/195035267-17306151-161e-4b75-9383-b55a08bfb915.png)

I am trying to read this file from 

this is the code snippet

> import pdfplumber as pdfp
> with pdfp.open("/content/drive/MyDrive/Books/Project Mgt/IT Portfolio Management.pdf") as temp:
>   pages = temp.pages
>   for page in pages:
>     print(page.extract_text())

**Comment by samkit-jain** [comment]
Hi @SridharVoleti Appreciate your interest in the library. If you are not getting the whitespace, you can try configuring the `x_tolerance` (defaults to 3) param in the `.extract_text(...)` method. It adds spaces where the difference between the `x1` of one character and the `x0` of the next is greater than `x_tolerance`.
